include/session/config/base.hpp

```diff
@@ -1098,6 +1208,29 @@ class ConfigBase : public ConfigSig {
     /// - `bool` -- Returns true if something has changed since last call to dump
     virtual bool needs_dump() const { return _needs_dump; }
 
+    /// API: base/ConfigBase::MULTIPART_MAX_WAIT
+    ///
+    /// This value controls how long we will store incomplete multipart messages since the last part
+    /// of such a message that we received, in the hopes of getting the rest of the parts soon.  The
+    /// default is a week: although long, this allows for extended downtime of a multidevice client
+    /// that uploads only some of the parts before going offline.
+    ///
+    /// Storage of such incomplete sets requires storing (via dump data) all the partial data until
+    /// the full set is received.
+    ///
+    /// Note that only *incomplete* partial sets are affected by this (and stored); we also
+    /// separately retain metadata about *completed* multipart sets (see MULTIPART_MAX_REMEMBER).
+    std::chrono::milliseconds MULTIPART_MAX_WAIT = 7 * 24h;
+
+    /// API: base/ConfigBase::MULTIPART_MAX_REMEMBER
+    ///
+    /// This value controls how long we retain the hashes of *completed* multipart config sets (so
+    /// that we can know to ignore duplicate message parts of messages we have already processed).
+    ///
+    /// Unlike MULTIPART_MAX_WAIT, such storage does not include the data itself, but merely the
+    /// (final) config hash and timestamp of the recombined parts needed for deduplication.
+    std::chrono::milliseconds MULTIPART_MAX_REMEMBER = 14 * 24h;
```

Comment:
Is this just an arbitrary duration on the off chance someones device goes offline part way through pushing up a multi-part config and doesn't come back online for a while?

(Just wondering why it's 14 days as opposed to 30 which is the default TTL for config messages)

Reply:
Yes.  14d was a fairly arbitrary choice, and it's really how long we expect to hold onto a set of incomplete chunks that another client uploaded before we give up on them and discard them.

30d would work fine, though; worst case we could just end up holding on to several 75kiB chunks of data for longer.

Reply:
No worries - I guess it probably doesn't matter _too_ much since if we swap the TTL extension logic to use `active_hashes()` then the parts of the config already on the swarm would stay alive for this expiration + 30 days anyway which should be more than enough time for the original device to come back online and complete the upload